backend/amd64/arch.mli

```diff
@@ -103,6 +103,8 @@ type specific_operation =
         locality: prefetch_temporal_locality_hint;
         addr: addressing_mode;
       }
+  | Illvm_intrinsic of string          (* Name of caml_* intrinsic (to be
+                                          lowered in llvmize) *)
```

Comment:
I find this comment is confusing, I would prefer to just have the CR in `arch.ml` about improving the type . 
```suggestion
  | Illvm_intrinsic of string         
```